### PR TITLE
bug fix: qwen-omni 4d attention mask

### PR DESCRIPTION
**Pull request by yuekaizhang**
# What does this PR do ?

When attention_mask's shape is 4D, we need the PR's fix.

Previous test case sets attention_mask=None, so we didn't find the error. 

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced attention mask tensor processing to ensure consistent and reliable behavior across different input configurations, improving model robustness.

### CODE REVIEW

File: src/megatron/bridge/models/qwen_omni/modeling_qwen25_omni/rope.py [hunk 1]
```diff
@@ -103,6 +103,12 @@ def get_rope_index(
         total_input_ids = input_ids
         if attention_mask is not None:
             attention_mask = attention_mask == 1
+            # Handle multi-dimensional attention masks (3D or 4D)
+            if attention_mask.dim() > 2:
+                attention_mask = attention_mask.any(dim=-1)
+                if attention_mask.dim() == 3:
+                    attention_mask = attention_mask.squeeze(1)
+                attention_mask = attention_mask.to(dtype=input_ids.dtype)
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply src/megatron/bridge/models/qwen_omni/modeling_qwen25_omni/rope.py]: @yuekaizhang  please check this ^^

**Comment by yaoyu-33** [comment]
@yuekaizhang  can you check ai comment?

**Review by yaoyu-33 — APPROVED** [review]

**Comment by pruprakash** [comment]
## QA RCCA Analysis

### 1. Fix Reference
- **PR**: #3086 - bug fix: qwen-omni 4d attention mask
- **Issue**: PR is the fix itself (no linked issue)

### 2. Root Cause
When attention_mask has 4D shape, the mask processing was incorrect. Previous test case set attention_mask=None, so the error wasn't caught.

### 3. Trigger Configuration
- Using Qwen-Omni model
- 4D attention mask inputs

### 4. Nature of the Bug
**Classification**: CODE BUG - Incorrect 4D attention mask handling

### 5. Existing Test Coverage
**In Fix PR**: NO - No test files (existing test used attention_mask=None)

**In NMFW Tests**: Not specifically covering 4D attention masks

### 6. Coverage Assessment
| Test Type | Exists | Covers Bug |
|-----------|--------|------------|
| Fix PR tests | NO | N/A |
| NMFW regression tests | NO | N/A |

### 7. New Regression Test
**NOT ADDED** - Model-specific fix. Existing tests should be updated to cover 4D mask case.

### 8. Conclusion
**Verdict**: PARTIAL COVERAGE - No dedicated test for 4D attention mask scenario.
